light_curves/lc_classifier.ipynb

```diff
@@ -437,72 +501,106 @@
+    "    dy = np.array(singleband_oid[\"err\"])\n",
     "    \n",
+    "    #kernel = 1.0 * RBF(length_scale=30)\n",
+    "    #gp = GaussianProcessRegressor(kernel=kernel, alpha=dy**2, normalize_y = False)\n",
+    "    #gp.fit(X, y)\n",
+    "    #mean_prediction,std_prediction = gp.predict(x_interpol, return_std=True)\n",
+    "\n",
+    "    #try KNN\n",
+    "    KNN = KNeighborsRegressor(n_neighbors = 3)\n",
+    "    KNN.fit(X, y)\n",
+    "    mean_prediction = KNN.predict(x_interpol)\n",
+    "        \n",
+    "    #KNN doesnt output an uncertainty array, so make our own:\n",
+    "    #an array of the same length as mean_prediction\n",
+    "    #having values equal to the mean of the original uncertainty array\n",
+    "    err = np.full_like(mean_prediction, singleband_oid.err.mean())  \n",
+    "        \n",
+    "    #get these values into the dataframe\n",
+    "    # append the results as a dict. the list will be converted to a dataframe later.\n",
+    "    lc_interpol.append(\n",
+    "        {\"objectid\": oid, \"label\": str(singleband_oid.label.unique().squeeze()), \"band\": band, \"time\": x_interpol.reshape(-1), \n",
+    "         \"flux\": mean_prediction, \"err\": err}\n",
+    "    )\n",
     "    \n",
-    "#output from this cell should show that all bands have the same length"
+    "        \n",
+    "    #se what this looks like on just a single light curve for now\n",
+    "    if (band == 'zr') and (oid == 9) :  \n",
+    "        #see if this looks reasonable\n",
+    "        plt.errorbar(X,y,dy,linestyle=\"None\",color=\"tab:blue\",marker=\".\")\n",
+    "        plt.plot(x_interpol, mean_prediction, label=\"Mean prediction\")\n",
+    "        #plt.fill_between(\n",
+    "        #    x_interpol.ravel(),\n",
+    "        #    mean_prediction - 1.96 * std_prediction,\n",
+    "        #    mean_prediction + 1.96 * std_prediction,\n",
+    "        #    color=\"tab:orange\",\n",
+    "        #    alpha=0.5,\n",
+    "        #    label=r\"95% confidence interval\",\n",
+    "        #)\n",
+    "        plt.legend()\n",
+    "        plt.xlabel(\"time\")\n",
+    "        plt.ylabel(\"flux\")\n",
+    "        _ = plt.title(\"KNN regression\")\n",
+    "        \n",
+    "        \n",
+    "# create a dataframe of the interpolated light curves\n",
+    "df_interpol = pd.DataFrame(lc_interpol)\n",
+    "# df_lc_interpol has one row per dict in lc_interpol. time and flux columns store arrays.\n",
+    "# \"explode\" the dataframe to get one row per light curve point. time and flux columns will now store floats.\n",
+    "df_lc = df_interpol.explode([\"time\", \"flux\",\"err\"], ignore_index=True)\n",
+    "\n",
+    "#somehow data types for some columns are coming in as 'object' which sktime doesn't like\n",
+    "#instead make them numeric data types\n",
+    "cols = [\"time\", \"flux\", \"err\"]\n",
+    "df_lc[cols] = df_lc[cols].apply(pd.to_numeric, errors='coerce')\n"
```

Comment:
Pandas `apply` can be very slow on large dataframes. I wonder if this could be done using either `df_lc. astype` or `df_lc.convert_dtypes`? I didn't take the time to check, just flagging it as something we probably want to look at in the future.

Reply:
I couldn't get`astype` to work, but didn't try `convert_dtypes`.